### PR TITLE
Stop unnecessary config duplication

### LINKED ISSUE #378: Add global remote_write to reduce reptitition
It would be nice to add a global `remote_write` section to the `prometheus` block to avoid repeating it multiple times across the file. This global block would be the default list of remote_write locations for integrations and all Prometheus instances. If `integrations.prometheus_remote_write` is defined, it overrides the global settings for integrations. If `remote_write` is defined in an instance config, it overrides the global settings for instances.

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description 

This adds a top level `remote write` block to prometheus that is used as a default to all subsequent blocks. If any of the subsequent blocks are defined then nothing happens. This ensures that it is backward compatible and allows the child to override the parent. 

Tested under normal usage and also in scraping service mode with hot reloading the configuration.

#### Which issue(s) this PR fixes 

Closes #378 

#### Notes to the Reviewer

Tested under both the normal modes of operation and scraping service.

#### PR Checklist

- [X] CHANGELOG updated 
- [X] Documentation added
- [X] Tests updated

### CODE REVIEW

File: pkg/prom/instance/instance.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ func (c Config) MarshalYAML() (interface{}, error) {
 // ApplyDefaults applies default configurations to the configuration to all
 // values that have not been changed to their non-zero value. ApplyDefaults
 // also validates the config.
-func (c *Config) ApplyDefaults(global *config.GlobalConfig) error {
+func (c *Config) ApplyDefaults(global *config.GlobalConfig, prometheusWriteConfig []*RemoteWriteConfig) error {
```
Comment (rfratto) [inline pkg/prom/instance/instance.go:117]: Can we call the param `defaultRemoteWrite` instead?

File: pkg/prom/instance/instance.go [hunk 2]
```diff
@@ -158,6 +158,11 @@ func (c *Config) ApplyDefaults(global *config.GlobalConfig) error {
 	}
 
 	rwNames := map[string]struct{}{}
+
+	// If the instance remote write is not filled in, then apply the prometheus write config
+	if len(c.RemoteWrite) == 0 && len(prometheusWriteConfig) > 0 {
```
Comment (rfratto) [inline pkg/prom/instance/instance.go:163]: Tiny nit: you don't need to check for `len(prometheusWriteConfig) > 0` :)

File: pkg/prom/instance/instance.go [hunk 3]
```diff
@@ -189,6 +194,8 @@ func (c *Config) ApplyDefaults(global *config.GlobalConfig) error {
 		rwNames[cfg.Base.Name] = struct{}{}
 	}
 
+
+
```
Comment (rfratto) [inline pkg/prom/instance/instance.go:198]: Did you mean to commit the extra newlines here?

File: pkg/prom/ha/server.go [hunk 4]
```diff
@@ -101,10 +101,12 @@ type Server struct {
 	exited chan bool
 
 	closeDependencies func() error
+
+	prometheusWriteConfig []*instance.RemoteWriteConfig
```
Comment (rfratto) [inline pkg/prom/ha/server.go:105]: Hmm, what do you think about making our own `GlobalConfig` struct in `pkg/prom/instance` and using that instead? I don't want to grow the list of params beyond what it is already:

```
type GlobalConfig struct {
  Prometheus *config.GlobalConfig 
  DefaultRemoteWrite []*RemoteWriteConfig
}
```

File: pkg/config/config.go [hunk 5]
```diff
@@ -56,6 +56,12 @@ func (c *Config) ApplyDefaults() error {
 		c.Integrations.ServerUsingTLS = c.Server.HTTPTLSConfig.TLSKeyPath != "" && c.Server.HTTPTLSConfig.TLSCertPath != ""
 	}
 
+	// The prometheus block remote write should trickle down to the integrations block if it is NOT defined.
+	promAndIntegrationsEnabled := c.Integrations.Enabled && c.Prometheus.Enabled
+	promHasRemoteIntegrationDoesNot := len(c.Prometheus.RemoteWrite) > 0 && len(c.Integrations.PrometheusRemoteWrite) == 0
+	if promAndIntegrationsEnabled && promHasRemoteIntegrationDoesNot {
+		c.Integrations.PrometheusRemoteWrite = c.Prometheus.RemoteWrite
+	}
```
Comment (rfratto) [inline pkg/config/config.go:64]: I think we can probably make some more assumptions here. If you move all of this inside `if c.IntegrationsEnabled` on line 53, you can just implement it as:

```go
if (c.Integrations.PrometheusRemoteWrite) == 0 {
  c.Integrations.PrometheusRemoteWrite = c.Prometheus.RemoteWrite 
}
```

**Review by rfratto — COMMENTED** [review]
Some tiny nits, looks good though :)

File: pkg/prom/instance/instance.go (hunk 1 above)
Comment (rfratto) [reply pkg/prom/instance/instance.go:117]: (No need if we go with a custom GlobalConfig struct, though)

File: pkg/prom/ha/server.go (hunk 4 above)
Comment (mattdurham) [reply pkg/prom/ha/server.go:105]: That has a larger blast radius than expected when I tried it. I would prefer to get what I have into main and then do a separate PR for that.

Comment (rfratto) [reply pkg/prom/ha/server.go:105]: SGTM

File: pkg/prom/ha/server.go [hunk 6]
```diff
@@ -101,10 +101,12 @@ type Server struct {
 	exited chan bool
 
 	closeDependencies func() error
+
+	defaultRemoteWrite []*instance.RemoteWriteConfig
 }
 
 // New creates a new HA scraping service instance.
-func New(reg prometheus.Registerer, cfg Config, globalConfig *config.GlobalConfig, clientConfig client.Config, logger log.Logger, im instance.Manager) (*Server, error) {
+func New(reg prometheus.Registerer, cfg Config, globalConfig *config.GlobalConfig, clientConfig client.Config, logger log.Logger, im instance.Manager, prometheusWriteConfig []*instance.RemoteWriteConfig) (*Server, error) {
```
Comment (rfratto) [inline pkg/prom/ha/server.go:109]: Should you rename it to defaultRemoteWrite here too?

**Review by rfratto — APPROVED** [review]
LGTM, thanks for working on this!

**Pull request merged**